src/PrestaShopBundle/Twig/Component/Head.php

```diff
@@ -75,6 +81,27 @@ public function getEmployeeToken(): string
 
     public function getJsDef(): array
     {
+        $jsDefs = [
+            'baseDir' => $this->context->getContext()->shop->getBaseURI(),
+            'baseAdminDir' => $this->context->getContext()->shop->getBaseURI() . basename(_PS_ADMIN_DIR_) . '/',
+            'currency' => [
+                'iso_code' => $this->context->getContext()->currency->iso_code,
+                'sign' => $this->context->getContext()->currency->symbol,
+                'name' => $this->context->getContext()->currency->name,
+                'format' => $this->context->getContext()->currency->format,
+            ],
+            'currency_specifications' => $this->preparePriceSpecifications(),
+            'number_specifications' => $this->prepareNumberSpecifications(),
+            'prestashop' => [
+                'debug' => $this->debugMode,
+            ],
+            'show_new_orders' => $this->configuration->get('PS_SHOW_NEW_ORDERS'),
+            'show_new_customers' => $this->configuration->get('PS_SHOW_NEW_CUSTOMERS'),
+            'show_new_messages' => $this->configuration->get('PS_SHOW_NEW_MESSAGES'),
+        ];
+
+        Media::addJsDef($jsDefs);
+
         return Media::getJsDef();
```

Comment:
I'm not sure we need to force addin the js defs, we should try and stop relying on the Media class at some point so at least for the variables we already handle correctly let's stop passing through it.

Reply:
```suggestion
        return array_merge([
            'baseDir' => $this->context->getContext()->shop->getBaseURI(),
            'baseAdminDir' => $this->context->getContext()->shop->getBaseURI() . basename(_PS_ADMIN_DIR_) . '/',
            'currency' => [
                'iso_code' => $this->context->getContext()->currency->iso_code,
                'sign' => $this->context->getContext()->currency->symbol,
                'name' => $this->context->getContext()->currency->name,
                'format' => $this->context->getContext()->currency->format,
            ],
            'currency_specifications' => $this->preparePriceSpecifications(),
            'number_specifications' => $this->prepareNumberSpecifications(),
            'prestashop' => [
                'debug' => $this->debugMode,
            ],
            'show_new_orders' => $this->configuration->get('PS_SHOW_NEW_ORDERS'),
            'show_new_customers' => $this->configuration->get('PS_SHOW_NEW_CUSTOMERS'),
            'show_new_messages' => $this->configuration->get('PS_SHOW_NEW_MESSAGES'),
        ], Media::getJsDef());
```